spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/InputBindingLifecycle.java

```diff
@@ -45,12 +46,13 @@ public void setApplicationContext(ApplicationContext applicationContext)
 	@Override
 	public void start() {
 		if (!running) {
+			ChannelBindingServiceProperties channelBindingServiceProperties = null;
 			// retrieve the ChannelBindingService lazily, avoiding early initialization
 			try {
 				ChannelBindingService channelBindingService = this.applicationContext
 						.getBean(ChannelBindingService.class);
-				Map<String, Bindable> bindables = this.applicationContext
-						.getBeansOfType(Bindable.class);
+				channelBindingServiceProperties = channelBindingService.getChannelBindingServiceProperties();
+				Map<String, Bindable> bindables = this.applicationContext.getBeansOfType(Bindable.class);
 				for (Bindable bindable : bindables.values()) {
 					bindable.bindInputs(channelBindingService);
```

Comment:
My concern was with performing the binding (starting the producer/consumer adapter contexts) even before the main application context is started. I guess that is fine since it is just the binding that is established.